test/integration/test_konnect_entities.go

```diff
@@ -143,6 +143,22 @@ func TestKonnectEntities(t *testing.T) {
 		assertKonnectEntityProgrammed(t, kpb)
 	}, testutils.ObjectUpdateTimeout, testutils.ObjectUpdateTick)
 
+	globalKPB := deploy.KongPluginBinding(t, ctx, clientNamespaced,
+		konnect.NewKongPluginBindingBuilder().
+			WithPluginRef(kp.Name).
+			WithControlPlaneRefKonnectNamespaced(cp.Name).
+			WithScope(configurationv1alpha1.KongPluginBindingScopeGlobalInControlPlane).
+			Build(),
+		deploy.WithTestIDLabel(testID),
+	)
+
+	t.Logf("Waiting for KongPluginBinding to be updated with Konnect ID")
+	require.EventuallyWithT(t, func(t *assert.CollectT) {
+		err := GetClients().MgrClient.Get(GetCtx(), types.NamespacedName{Name: globalKPB.Name, Namespace: globalKPB.Namespace}, globalKPB)
+		require.NoError(t, err)
```

Comment:
I believe this is the problematic use case which can panic when `err != nil`, right? AFAIR we want to use `assert` for these, like so

```suggestion
		assert.NoError(t, err)
```

Reply:
Not really. It was fixed in https://github.com/stretchr/testify/pull/1481. Now it's fine to use `require` inside of `EventuallyWithT` as it will just exit a goroutine running a check. :)

Reply:
Oh, good to know 👍